trends_presentation/src/commonMain/kotlin/net/thechance/mena/trends/presentation/screen/update_categories/UpdateCategoriesScreen.kt

```diff
@@ -0,0 +1,202 @@
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import mena.trends_presentation.generated.resources.Res
+import mena.trends_presentation.generated.resources.back_arrow
+import mena.trends_presentation.generated.resources.change_tags
+import mena.trends_presentation.generated.resources.choose_interests
+import mena.trends_presentation.generated.resources.help_text
+import mena.trends_presentation.generated.resources.ic_arrow_left
+import mena.trends_presentation.generated.resources.save_change
+import net.thechance.mena.designsystem.presentation.component.appBar.AppBar
+import net.thechance.mena.designsystem.presentation.component.button.PrimaryButton
+import net.thechance.mena.designsystem.presentation.component.icon.Icon
+import net.thechance.mena.designsystem.presentation.component.indicator.DotsProgressIndicator
+import net.thechance.mena.designsystem.presentation.component.scaffold.Scaffold
+import net.thechance.mena.designsystem.presentation.component.text.Text
+import net.thechance.mena.designsystem.presentation.theme.theme.MenaTheme
+import net.thechance.mena.designsystem.presentation.theme.theme.Theme
+import net.thechance.mena.trends.presentation.navigation.LocalNavController
+import net.thechance.mena.trends.presentation.navigation.Route
+import net.thechance.mena.trends.presentation.shared.base.toStringResource
+import net.thechance.mena.trends.presentation.shared.component.CategoryItem
+import net.thechance.mena.trends.presentation.shared.component.snackbar.TrendsSnackBar
+import net.thechance.mena.trends.presentation.shared.model.SnackBarStatus
+import net.thechance.mena.trends.presentation.shared.util.ObserveAsEffect
+import org.jetbrains.compose.resources.painterResource
+import org.jetbrains.compose.resources.stringResource
+import org.jetbrains.compose.ui.tooling.preview.Preview
+import org.koin.compose.viewmodel.koinViewModel
+
+@Composable
+internal fun UpdateCategoriesScreen(
+    viewModel: UpdateCategoriesViewModel = koinViewModel(),
+) {
+    val state by viewModel.state.collectAsStateWithLifecycle()
+    val navController = LocalNavController.current
+
+    ObserveAsEffect(effects = viewModel.effect) { effect ->
+        when (effect) {
+            is UpdateCategoriesScreenEffect.NavigateBack -> navController.popBackStack()
+            is UpdateCategoriesScreenEffect.NavigateToTrends -> navController.navigate(Route.Trends)
+        }
+    }
+
+    LaunchedEffect(Unit) {
+        viewModel.getCategories()
+    }
+
+    CategoryPickScreenContent(
+        state = state,
+        listener = viewModel
+    )
+}
+
+@Composable
+private fun CategoryPickScreenContent(
```

Comment:
I think this name need to changed to 
```suggestion
private fun UpdateCategoryScreenContent(
```

Reply:
Done.